pennylane/ops/op_math/composite.py

```diff
@@ -27,6 +28,23 @@
 # pylint: disable=too-many-instance-attributes
 
 
+def handle_recursion_error(func):
+    """Handles any recursion errors raised from too many nested products."""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except RecursionError as e:
+            raise RuntimeError(
+                "Maximum recursion depth reached! This is likely due to nesting too many "
+                "levels of composite operators. Try setting lazy=False when calling qml.sum "
+                "and qml.prod, or use the + and @ operators instead."
```

Comment:
```suggestion
                "levels of composite operators. Try setting lazy=False when calling qml.sum "
                "and qml.prod, or use the + and @ operators instead, or periodically call qml.simplify on "
                "your operators. "
```
or something better phrased than that.